plugin-api/src/main/java/org/hyperledger/besu/plugin/data/BlockPropagated.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.plugin.data;
+
+import org.hyperledger.besu.plugin.Unstable;
+
+import java.util.function.Supplier;
+
+/** The minimum set of data for a BlockPropagated. */
+@Unstable
+public interface BlockPropagated {
+
+  /**
+   * A {@link BlockHeader} object.
+   *
+   * @return A {@link BlockHeader}
+   */
+  BlockHeader getBlockHeader();
+
+  /**
+   * A scalar value corresponding to the total difficulty.
+   *
+   * @return A scalar value corresponding to the total difficulty.
+   */
+  Quantity getTotalDifficulty();
+
+  static BlockPropagated of(
```

Comment:
I'm going to suggest that this `static` constructor of anonymous `BlockPropagated` objects should be moved to the place it is being used ...or perhaps use a POJO rather than an interface?

Reply:
Moved it to the place it is being used.
See commit https://github.com/hyperledger/besu/pull/97/commits/46f84c6db08a7d07c24773fbc6069fc49eb8a097